src/browser_app_skeleton/src/actions/browser_action.cr.ecr

```diff
@@ -25,7 +25,8 @@ abstract class BrowserAction < Lucky::Action
   expose current_user
 
   # This method tells Authentic how to find the current user
-  private def find_current_user(id) : User?
+  # The 'memoize' macro makes sure only one query is issued to find the user
+  private memoize def find_current_user(id) : User?
```

Comment:
You can just set this type to `User`. If it's nil, it just won't memoize. I saw you opened the issue already for that, but until we have that in, this could be a small win.

Reply:
I'm not sure I understand. The User can be nil so I can't change the return type to not have nil.

Reply:
The `memoize` right now doesn't allow you to specify a union like `User?` https://github.com/luckyframework/lucky/blob/master/src/lucky/memoizable.cr#L21 so this should throw a compile time error.

Reply:
Yeah this does throw an error so I opened the issue to add support. I’m not sure how we’d get a quick win by setting it to User because that also won’t compile. I am quite tired so maybe just being dense haha

Reply:
Now that memoize supports taking args and having a nil value, I think this is fine. 😄